naga/src/front/wgsl/parse/conv.rs

```diff
@@ -252,3 +257,26 @@ pub fn map_conservative_depth(
         _ => Err(Error::UnknownConservativeDepth(span)),
     }
 }
+
+pub fn map_subgroup_operation(
+    word: &str,
+) -> Option<(crate::SubgroupOperation, crate::CollectiveOperation)> {
+    use crate::CollectiveOperation as co;
+    use crate::SubgroupOperation as sg;
+    Some(match word {
+        "subgroupAll" => (sg::All, co::Reduce),
+        "subgroupAny" => (sg::Any, co::Reduce),
+        "subgroupAdd" => (sg::Add, co::Reduce),
+        "subgroupMul" => (sg::Mul, co::Reduce),
+        "subgroupMin" => (sg::Min, co::Reduce),
+        "subgroupMax" => (sg::Max, co::Reduce),
+        "subgroupAnd" => (sg::And, co::Reduce),
+        "subgroupOr" => (sg::Or, co::Reduce),
+        "subgroupXor" => (sg::Xor, co::Reduce),
+        "subgroupPrefixExclusiveAdd" => (sg::Add, co::ExclusiveScan),
+        "subgroupPrefixExclusiveMul" => (sg::Mul, co::ExclusiveScan),
+        "subgroupPrefixInclusiveAdd" => (sg::Add, co::InclusiveScan),
+        "subgroupPrefixInclusiveMul" => (sg::Mul, co::InclusiveScan),
```

Comment:
```suggestion
        "subgroupSum" => (sg::Add, co::Reduce),
        "subgroupProduct" => (sg::Mul, co::Reduce),
        "subgroupMin" => (sg::Min, co::Reduce),
        "subgroupMax" => (sg::Max, co::Reduce),
        "subgroupAnd" => (sg::And, co::Reduce),
        "subgroupOr" => (sg::Or, co::Reduce),
        "subgroupXor" => (sg::Xor, co::Reduce),
        "subgroupExclusiveSum" => (sg::Add, co::ExclusiveScan),
        "subgroupExclusiveProduct" => (sg::Mul, co::ExclusiveScan),
        "subgroupInclusiveSum" => (sg::Add, co::InclusiveScan),
        "subgroupInclusiveProduct" => (sg::Mul, co::InclusiveScan),
```

We should keep the naming in sync with the [upstream proposal](https://github.com/gpuweb/gpuweb/blob/main/proposals/subgroups.md).

Reply:
WGSL has always copied the GLSL naming scheme. Thus I think we should stick with "Add" and "Mul" and change the upstream proposal instead. I removed the "Prefix" though as that is not present in GLSL.